src/main/webapp/app/exercises/programming/shared/instructions-render/extensions/programming-exercise-plant-uml.extension.ts

```diff
@@ -92,11 +101,16 @@ export class ProgrammingExercisePlantUmlExtensionWrapper implements ArtemisShowd
                 // before we send the plantUml to the server for rendering, we need to inject the current test status so that the colors can be adapted
                 // (green == implemented, red == not yet implemented, grey == unknown)
                 const plantUmlsValidated = plantUmlsIndexed.map((plantUmlIndexed: { plantUmlId: number; plantUml: string }) => {
-                    plantUmlIndexed.plantUml = plantUmlIndexed.plantUml.replace(/testsColor\(((?:[^()]+\([^()]*\))*[^()]*)\)/g, (match: any, capture: string) => {
-                        // split the names by "," only when there is not a closing bracket without a previous opening bracket
-                        const tests = capture.split(/,(?![^(]*?\))/);
+                    plantUmlIndexed.plantUml = plantUmlIndexed.plantUml.replace(this.testsColorRegex, (match: any, capture: string) => {
+                        const tests = this.programmingExerciseInstructionService.convertTestListToIds(capture, this.testCases);
                         const { testCaseState } = this.programmingExerciseInstructionService.testStatusForTask(tests, this.latestResult);
-                        return testCaseState === TestCaseState.SUCCESS ? 'green' : testCaseState === TestCaseState.FAIL ? 'red' : 'grey';
+                        if (testCaseState === TestCaseState.SUCCESS) {
+                            return 'green';
+                        } else if (testCaseState === TestCaseState.FAIL) {
+                            return 'red';
+                        } else {
+                            return 'grey';
+                        }
```

Comment:
Consider unsing a switch-case
Feels smoother when working with enums. Just my personal opinion. Leave it if you don't agree.